### PR TITLE
Lib refactoring

### PR DESCRIPTION
**Pull request by LeasoftKonstantinShapoval**

### CODE REVIEW

**Review by tiutiunnykleasoft — CHANGES_REQUESTED** [review]
If Afterpay switched off other payment doesnt works. 
Also on return url need corrections

**Comment by tiutiunnykleasoft** [comment]
1. in `plgVmConfirmedOrder` method: $cart is not using - should remove it?.
2. for method `usersFromAllowedCountries` method must be updated PHPcomments.
3. The error with unworked payments when Afterpay is switched off is still present.
4. https://github.com/emspay/ems-online-virtuemart/pull/13#discussion_r497300909

**Comment by LeasoftKonstantinShapoval** [comment]
@tiutiunnykleasoft  Re review please.

File: libraries/emspay/Emspay/Lib/EmspayVmPaymentPlugin.php [hunk 1]
```diff
@@ -239,7 +239,25 @@ protected function getOrderIdByGingerOrder($gingerOrderId)
         }
         return 0;
     }
-    
+
+    /**
+     * fetch vm order number form hte payment table
```
Comment (tiutiunnykleasoft) [inline libraries/emspay/Emspay/Lib/EmspayVmPaymentPlugin.php:244]: drunk comment

File: plugins/vmpayment/emspayafterpay/emspayafterpay.php [hunk 2]
```diff
@@ -195,8 +195,10 @@ protected function gettermsAndConditionsUrlByCountry($country)
      */
     public function plgVmOnCheckoutCheckDataPayment(VirtueMartCart $cart)
     {
-        if(!$this->userIsFromAllowedCountries($cart->BTaddress['fields']['virtuemart_country_id']['country_2_code'])) {
-            return false;
+        if(!$cart->cartData['paymentName'] = 'AfterPay'){
```
Comment (tiutiunnykleasoft) [inline plugins/vmpayment/emspayafterpay/emspayafterpay.php:198]: Hint:
`=` declare the variable.
`==` compares values.

**Review by tiutiunnykleasoft — COMMENTED** [review]
fix

**Comment by LeasoftKonstantinShapoval** [comment]
Fixed and refactored.

**Review by tiutiunnykleasoft — CHANGES_REQUESTED** [review]
Mostly OK. But still, see order number of finish page

**Comment by LeasoftKonstantinShapoval** [comment]
order_id removed

**Comment by LeasoftKonstantinShapoval** [comment]
Updated

File: plugins/vmpayment/emspayamex/emspayamex.php [hunk 3]
```diff
@@ -152,7 +152,7 @@ public function plgVmConfirmedOrder($cart, $order)
      * @return bool|null|string
      * @since v1.0.0
      */
-    public function plgVmOnPaymentResponseReceived(&$virtuemart_order_id, &$html)
+    public function plgVmOnPaymentResponseReceived(&$html, &$payment_reponse)
```
Comment (tiutiunnykleasoft) [inline plugins/vmpayment/emspayamex/emspayamex.php:155]: Update PHPdoc in this and other payments

File: plugins/vmpayment/emspaypaypal/emspaypaypal.php [hunk 4]
```diff
@@ -177,7 +177,7 @@ public function plgVmOnPaymentResponseReceived(&$virtuemart_order_id, &$html)
         $html = "<p>" . EmspayHelper::getOrderDescription($virtuemart_order_number) . "</p>";
         if ($statusSucceeded) {
             $this->emptyCart(null, $virtuemart_order_id);
-            $html .= "<p>". JText::_('EMSPAY_LIB_THANK_YOU_FOR_YOUR_ORDER'). "</p>";
```
Comment (tiutiunnykleasoft) [inline plugins/vmpayment/emspaypaypal/emspaypaypal.php:179]: Remove `EMSPAY_LIB_THANK_YOU_FOR_YOUR_ORDER` from the languages translation files

**Comment by LeasoftKonstantinShapoval** [comment]
@tiutiunnykleasoft Do we need add some notification to Changelog?

**Comment by tiutiunnykleasoft** [comment]
> @tiutiunnykleasoft Do we need add some notification to Changelog?

@LeasoftKonstantinShapoval yes

**Comment by LeasoftKonstantinShapoval** [comment]
> @LeasoftKonstantinShapoval yes

Something like "Fixes bug with wrong call after return url."?

**Comment by LeasoftKonstantinShapoval** [comment]
Updated.

**Comment by tiutiunnykleasoft** [comment]
Great work!
Actually, we must display the "Thank you message" only if payment completed! As I can see in `apple-pay` and other payment, when order returned with "new" status, we display "Thank you page", is it correct for you?
At this moment I can't approve changes until we do not end the discussion of that question.

**Comment by LeasoftKonstantinShapoval** [comment]
> As I can see in apple-pay and other payment, when order returned with "new" status, we display "Thank you page", is it correct for you?
Yes. it is.

**Comment by tiutiunnykleasoft** [comment]
> > As I can see in apple-pay and other payment, when order returned with "new" status, we display "Thank you page", is it correct for you?
> 
> Yes. it is.

In this case, what do we thank the user for?)

**Comment by LeasoftKonstantinShapoval** [comment]
For patience. :)

**Comment by tiutiunnykleasoft** [comment]
> For patience. :)

Good idea :)
Could the user have misunderstandings with the store due to the message
"Thank you for your Order!
Paid with аpple-pay "?

When in fact he changed his mind or was unable to pay.

**Comment by LeasoftKonstantinShapoval** [comment]
> Could the user have misunderstandings with the store due to the message
> "Thank you for your Order!
> Paid with аpple-pay "?

Of course, and this bug stands not only in VirtueMart. But earlier I ask for this and got answer 'Its a feature'.

**Comment by tiutiunnykleasoft** [comment]
> > Could the user have misunderstandings with the store due to the message
> > "Thank you for your Order!
> > Paid with аpple-pay "?
> 
> Of course, and this bug stands not only in VirtueMart. But earlier I ask for this and got answer 'Its a feature'.

Can you send a link to this question if you has ?

**Comment by LeasoftKonstantinShapoval** [comment]
> Can you send a link to this question if you has ?

It was in skype or in voice chat, or in office.

**Comment by LeasoftKonstantinShapoval** [comment]
We can start new task and fix this.
What do you think?

**Comment by LeasoftKonstantinShapoval** [comment]
` case 'new':
                $this->updateOrderStatus($this->methodParametersFactory()->statusNew(), $virtuemart_order_id);
                return true;`
https://github.com/emspay/ems-online-virtuemart/blob/8b5efb7cbe0857873d90c41b9a5713017671a5b8/libraries/emspay/Emspay/Lib/EmspayVmPaymentPlugin.php#L192

I think its happening because we got status 'new' as true in status response.

**Comment by tiutiunnykleasoft** [comment]
> `case 'new': $this->updateOrderStatus($this->methodParametersFactory()->statusNew(), $virtuemart_order_id); return true;`
> 
> https://github.com/emspay/ems-online-virtuemart/blob/8b5efb7cbe0857873d90c41b9a5713017671a5b8/libraries/emspay/Emspay/Lib/EmspayVmPaymentPlugin.php#L192
> 
> I think its happening because we got status 'new' as true in status response.

Look like it.

**Review by tiutiunnykleasoft — APPROVED** [review]

**Pull request merged**